backend/forms/handlers/form.py

```diff
@@ -0,0 +1,33 @@
+from django.db.models.signals import post_save, pre_save
+from django.dispatch import receiver
+
+from core.utils.model_utils import slugify
+
+from ..models.form import Form
+
+
+@receiver(pre_save, sender=Form)
+def form_pre_save(sender, instance: Form, **kwargs):
+    if instance.slug is None:
+        instance.slug = slugify(instance.title)
+
+
+@receiver(post_save, sender=Form)
+def form_post_save(sender, instance: Form, *, created: bool, update_fields: list[str] | None, **kwargs):
+    """
+    We can't call build_enriched_fields on an unsaved form.
+    This will cause a second save, but forms are changed seldom so this should
+    not cause a performance issue.
+    Explicit update_fields=["enriched_fields"] is used to signal that the
```

Comment:
Comment mismatches implementation.

Reply:
Corrected `enriched_fields` to `cached_enriched_fields`, see [Slack](https://con2.slack.com/archives/C3ZGNGY48/p1705831755991039) for background